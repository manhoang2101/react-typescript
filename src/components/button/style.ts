import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    appButton: {
      "&:hover": {
        backgroundColor: "#fff",
        borderColor: "#fff",
        boxShadow: "none",
      },
      "&:active": {
        boxShadow: "none",
        backgroundColor: "#0062cc",
        borderColor: "#005cbf",
      },
      "&:focus": {
        boxShadow: "0 0 0 0.2rem rgba(0,123,255,.5)",
      },
    },
  });

export default style;
