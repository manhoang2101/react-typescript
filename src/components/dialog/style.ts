import { Theme, createStyles } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    form: {
      display: "flex",
      flexDirection: "column",
      margin: "auto",
      width: "fit-content",
    },
    formControl: {
      marginTop: theme.spacing(2),
      minWidth: 120,
    },
    formControlLabel: {
      marginTop: theme.spacing(1),
    },
    appBar: {
      position: "relative",
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
    title: {
      marginLeft: 0,
      flex: 1,
    },
    gutters: {
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
  });

export default style;
