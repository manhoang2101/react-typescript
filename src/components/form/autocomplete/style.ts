import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    TextField: {
      paddingRight: 25,
    },

    MuiAutocomplete: {
      input: {
        paddingRight: 25,
      },
    },
    KeyboardArrowDownIcon: {
      paddingLeft: 0,
      borderLeft: `1px solid #ccc`,
    },
  });

export default style;
