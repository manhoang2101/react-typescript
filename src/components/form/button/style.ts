import { createStyles } from "@material-ui/core";

const style = () => createStyles({});

export default style;
