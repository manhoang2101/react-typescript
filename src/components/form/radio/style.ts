import { createStyles, Theme } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    MuiFormHelperText: {
      marginLeft: 0,
    },
  });

export default style;
