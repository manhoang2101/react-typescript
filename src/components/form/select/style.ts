import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    formControl: {
      minWidth: 120,
    },
  });

export default style;
