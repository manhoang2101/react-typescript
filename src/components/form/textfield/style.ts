import { createStyles } from "@material-ui/core";

const style = () =>
  createStyles({
    AppTextField: {
      display: "block",
    },
  });

export default style;
