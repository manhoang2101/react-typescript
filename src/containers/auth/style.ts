import { Theme, createStyles } from "@material-ui/core";

const style = (theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      padding: theme.spacing(2),
      textAlign: "center",
      color: theme.palette.text.secondary,
    },
  });

export default style;
