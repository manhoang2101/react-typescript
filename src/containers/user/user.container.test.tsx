import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import { withStyles } from "@material-ui/core";
import { render } from "@testing-library/react";
import UserContainer from ".";
import style from "src/pages/user/style";
const mockStore = configureStore([]);
describe("My Connected React-Redux MasterContainer", () => {
  let store: any;
  let warper: any;
  let Component: any;
  beforeEach(() => {
    store = mockStore({
      myState: "sample text",
    });
    Component = withStyles(style)(UserContainer);
    store.dispatch = jest.fn();
    warper = render(
      <Provider store={store}>
        <BrowserRouter>
          <Component />
        </BrowserRouter>
      </Provider>
    );
  });
  it("should render with given state from Redux store", () => {
    expect(warper).toMatchSnapshot();
  });

  it("should onSelectUser", () => {
    const setUser = jest.fn();
    const container = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <Component
            fetchUsersAction={jest.fn}
            users={[]}
            user={null}
            setUser={setUser}
          />
        </BrowserRouter>
      </Provider>
    )
      .find(Component)
      .dive()
      .dive()
      .instance();
    container.onSelectUser({
      id: 1,
      name: "ABC",
    });
    expect(setUser).toBeCalled();
  });
  it("should addUser", () => {
    const addUser = jest.fn();
    const container = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <Component
            fetchUsersAction={jest.fn}
            users={[]}
            user={null}
            addUser={addUser}
          />
        </BrowserRouter>
      </Provider>
    )
      .find(Component)
      .dive()
      .dive()
      .instance();
    container.addUser({
      id: 1,
      name: "ABC",
    });
    expect(addUser).toBeCalled();
  });
});
