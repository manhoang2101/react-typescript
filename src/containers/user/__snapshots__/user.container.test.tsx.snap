// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux MasterContainer should render with given state from Redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          aria-label="simple"
          class="MuiTable-root AppTable-table-2 UserContainer-table-1"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root tr-thear MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head id AppTable-tdThead-6 td-thear"
                scope="col"
              >
                id
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head name AppTable-tdThead-6 td-thear"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head cardNumber AppTable-tdThead-6 td-thear"
                scope="col"
              >
                cardNumber
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head cardType AppTable-tdThead-6 td-thear"
                scope="col"
              >
                cardType
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple"
        class="MuiTable-root AppTable-table-2 UserContainer-table-1"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root tr-thear MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head id AppTable-tdThead-6 td-thear"
              scope="col"
            >
              id
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head name AppTable-tdThead-6 td-thear"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head cardNumber AppTable-tdThead-6 td-thear"
              scope="col"
            >
              cardNumber
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head cardType AppTable-tdThead-6 td-thear"
              scope="col"
            >
              cardType
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
